controllers/notes.js

```diff
@@ -0,0 +1,99 @@
+const note = require('../models/note');
+const { StatusCodes } = require('http-status-codes');
+const parseVErr = require('../utils/parseValidationErr');
+
+const getAllNotes = async (req, res) => {
+  const notes = await note.find({ createdBy: req.user.id }).sort('CreatedAt');
+  res.status(StatusCodes.OK).render('notes', { notes });
+};
+
+const addNote = async (req, res) => {
+  req.body.createdBy = req.user.id;
+  await note.create(req.body);
+  res.status(StatusCodes.CREATED);
+
+  const notes = await note.find({ createdBy: req.user.id }).sort('CreatedAt');
+  const info = [`The note has been successfully added.`];
+  res.render('notes', { notes, info });
+};
+
+const addForm = async (req, res) => {
+  res.render('note', { note: null });
+};
+
+const editForm = async (req, res) => {
+  const note = await note.findOne({
+    _id: req.params.id,
+    createdBy: req.user.id,
+  });
+
+  if (!note) {
+    throw new Error(`No note with id ${req.params.id}`);
+  }
+  res.status(StatusCodes.OK).render('note', { note });
+};
+
+const editNote = async (req, res) => {
+  const {
+    body: { title, subject, content },
+    user: { id: userId },
+    params: { id: noteId },
+  } = req;
+
+  if (title === '' || subject === '') {
```

Comment:
If the title, subject, or content are empty, you must display an error using the req.flash method and redirect to the '/notes/edit' route. You don't need to find the note because it's will be done by the editForm method.
```
if (!title || !subject || !content) {
    req.flash('error', "You must include the title, subject and content.");
    return res.redirect(`/notes/edit/${noteId}`);
  }
```